includes/Classifai/Admin/Metabox.php

```diff
@@ -0,0 +1,149 @@
+<?php
+/**
+ * Metabox class.
+ *
+ * @since 2.2.3
+ * @package Classifai
+ */
+
+namespace Classifai\Admin;
+
+use Classifai\Providers\OpenAI\Embeddings;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * This class generates the metabox for the classic editor post types.
+ */
+class Metabox {
+
+	/**
+	 * Current post type name.
+	 *
+	 * @var string
+	 */
+	protected $post_type = '';
+
+	/**
+	 * @var \Classifai\Providers\OpenAI\Embeddings
+	 */
+	private $embeddings;
+
+	/**
+	 * Checks whether this class's register method should run.
+	 *
+	 * @return bool
+	 * @since 1.4.0
+	 */
+	public function can_register() {
+		return is_admin();
+	}
+
+	/**
+	 * Class constructor.
+	 */
+	public function register() {
+		$this->embeddings = new Embeddings( false );
+		add_action( 'add_meta_boxes', [ $this, 'add_metabox' ] );
+		add_action( 'save_post', [ $this, 'save_metabox' ] );
+	}
+
+	/**
+	 * Add metabox.
+	 *
+	 * @param string $post_type Post type name.
+	 */
+	public function add_metabox( $post_type ) {
+
+		$this->post_type = $post_type;
+		$settings        = $this->embeddings->get_settings();
+
+		// Set up the embeddings metabox if the feature is enabled.
+		if ( isset( $settings['enable_classification'] ) && 1 === (int) $settings['enable_classification'] ) {
+			$embeddings_post_types = $this->embeddings->supported_post_types();
+			$classic_post_types    = get_post_types( [ 'show_in_rest' => false ], 'names', 'and' );
+			$post_types            = apply_filters( 'classifai_embeddings_classic_types', array_intersect( $embeddings_post_types, $classic_post_types ) );
```

Comment:
Can we add a filter docblock here, so this gets added to our documentation automatically?